### PR TITLE
style(black): add black formatter CI/pre-commit, formatted existing codes

### LINKED ISSUE #15: Set up the pre-commit workflow for automatic formatting on code style
Maybe we can add a formatter like black in the Makefile and gitaction to make life easier for developer.
What do you think?

### PR DESCRIPTION
**Pull request by tczhao**
resolves #15 

Added black formatter in CI, 
Added pre-commit hook to auto-format each commit (see CONTRIBUTING.md for installation step)
formatted existing codes

### CODE REVIEW

**Comment by xuyxu** [comment]
Great, thanks for the contribution @tczhao! I will take a look at the changes as soon as possible.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+repos:
+-   repo: https://github.com/ambv/black
+    rev: 20.8b1
+    hooks:
+    - id: black
+      language_version: python3
```
Comment (xuyxu) [inline .pre-commit-config.yaml:6]: I have relaxed the `language_version` from `python3.7` to `python3`. Let me know if there is any problem, and I will merge this PR if it's fine.

Comment (xuyxu) [reply .pre-commit-config.yaml:6]: @tczhao

**Comment by xuyxu** [comment]
In addition, I think we may need to provide new contributors with more detailed instructions on the entire workflow. I will work on this and set up a new page named `How to Contribute`, similar to [this](https://www.sktime.org/en/latest/contributing.html).

File: .pre-commit-config.yaml (hunk 1 above)
Comment (tczhao) [reply .pre-commit-config.yaml:6]: @xuyxu yes, works fine

**Comment by tczhao** [comment]
> In addition, I think we may need to provide new contributors with more detailed instructions on the entire workflow. I will work on this and set up a new page named `How to Contribute`, similar to [this](https://www.sktime.org/en/latest/contributing.html).

Yes, great idea, need to convert that contributing thing to rst as well

**Pull request merged**